---
layout: page
permalink: cshl.html
---

<!-- -->
<div class='section'>
  <div class='section-title'>Advanced Sequencing Technologies and Applications: BAM tutorial</div>
  <div class='description'>
    The goal of this page is to highlight a number of tools that can be used to view and manipulate sequence alignment (BAM) files. These tools are all included in the gkno package, so no additional tools need to be downloaded to work through these exercises.
  </div>

  <!-- Preparation -->
  <div class='section-subtitle'>Set up</div>
  <div class='description'>
    To begin, create a new directory call 'bam-demo'.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>cd workspace</li>
      <li>mkdir bam-demo</li>
      <li>cd bam-demo</li>
    </ul>
  </div>

  <!-- tools -->
  <div class='section-subtitle'>bamtools and samtools</div>
  <div class='description'>
    We will be using <a href="https://github.com/samtools/samtools.git"><i>samtools</i></a> for this tutorial. This is the most common tool for working with BAM files and is well maintained and trusted. Other tools exist, but this will suffice for the majority of tasks.
  </div>

  <!-- samtools help -->
  <div class='section-subtitle'>samtools help</div>
  <div class='description'>
    To see help for samtools, type:
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools</li>
    </ul>
  </div>
  <div class='description'>
    You can see that there are a lot of subcommands for <i>samtools</i>. Each of these performs a different task, some of which we will go through in this tutorial. To get help on individual subcommands, simply include the subcommand you want to know about.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools view</li>
      <li>samtools sort</li>
    </ul>
  </div>
  <div class='description'>
    The subcommand help gives you the format of the command line expected by <i>samtools</i>, including the input and output files as well as a list of all of the additional arguments that can be selected along with a brief description of their role. We can see more detailed help with the following command:
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools view -?</li>
    </ul>
  </div>

  <!-- samtools view -->
  <div class='section-subtitle'>samtools view</div>
  <div class='description'>
    The first thing that we probably want to do with the BAM file is look at it. Since the BAM file is a compressed binary format, using standard bash commands like 'more' or 'less', won't work.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools view sample.bam | less</li>
    </ul>
  </div>
  <div class='description'>
    The <span class="code-list code-span panel panel-danger">|</span> character 'pipes' the output from the <i>samtools</i> command into the bash command. The command <span class="code-list code-span panel panel-danger">less</span> prints the results of the <i>samtools</i> command to screen. This is really difficult to read, since the lines are longer than the screen width and 'wrap' around. Press 'q' to terminate this view, and try the same command as above, but include <span class="code-list code-span panel panel-danger">-S</span>. This makes the lines far easier to read.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools view sample.bam | less -S</li>
    </ul>
  </div>
  <div class='description'>
    You can type <span class="code-list code-span panel panel-danger">-S</span> and press return while the <i>less</i> view is open and you can toggle between these two views.
  </div>

  <!-- BAM header -->
  <div class='section-subtitle'>The BAM header</div>
  <div class='description'>
    You probably noticed that we didn't see the BAM header in the previous views. It is there, we just need to tell <i>samtools</i> that we want to see it.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools view -H sample.bam | less -S</li>
    </ul>
  </div>
  <div class='description'>
    Including <span class="code-list code-span panel panel-danger">-H</span> now tells <i>samtools</i> to only view the header and so this is all we see. If we instead use <span class="code-list code-span panel panel-danger">-h</span>, we will see the whole BAM file, header, alignments and all.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools view -h sample.bam | less -S</li>
    </ul>
  </div>

  <!-- samtools sort -->
  <div class='section-subtitle'>samtools sort</div>
  <div class='description'>
    You may have also noticed that the BAM file is not sorted. Aligners typically take FASTQ files and try to align them to the reference sequence. The resulting BAM file contains all of the alignments in the same order that they appeared in the FASTQ file, which is completely random in terms of genomic coordinates. For most downstream tools (e.g. <a href="http://iobio.io"><i>iobio</i></a>, <a href="https://www.broadinstitute.org/igv/"><i>IGV</i></a> or variant callers), the BAM file needs to be sorted by genomic coordinates. We can easily do this:
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools sort -o sample_sorted.bam sample.bam</li>
    </ul>
  </div>
  <div class='description'>
    <i>Samtools</i> only needs us to specify the input and output file names using <span class="code-list code-span panel panel-danger">-o</span>. We can use the <span class="code-list code-span panel panel-danger">-O</span> option to define the format of the output file, e.g. SAM, BAM, or CRAM. Now let's take a look at the new BAM file and check that it's sorted properly.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools view -h sample_sorted.bam | less -S</li>
    </ul>
  </div>
  <div class='description'>
    You can see that the alignments are now in coordinate order, but you can also see that the first line in the BAM header states (with the <span class="code-list code-span panel panel-danger">SO</span> tag in the <span class="code-list code-span panel panel-danger">@HD</span> line), that the Sort Order is 'coordinate' as required.
  </div>

  <!-- samtools index -->
  <div class='section-subtitle'>samtools index</div>
  <div class='description'>
    Now we have a sorted BAM file, but this usually isn't enough for most tools. For example, both <a href="http://iobio.io"><i>iobio</i></a> and <a href="https://www.broadinstitute.org/igv/"><i>IGV</i></a> want to look at specific regions within the BAM file and don't want to have to search through the entire file to find it, since the search would take an extremely long time. To achieve this, BAM files need to be indexed. This index provides tools a map of the data in the BAM file, so that they can quickly jump to the regions of interest. So, let's index our BAM file:
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools index sample_sorted.bam</li>
    </ul>
  </div>
  <div class='description'>
    If we look in our directory (<span class="code-list code-span panel panel-danger">ls</span>), we will see the index file with the extension '.bai'. Let's exploit this index to see the reads that fall into the region '20000-30000' on chromosome 20.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools view sample_sorted.bam 20:20000-30000 | less -S</li>
    </ul>
  </div>
  <div class='description'>
    What if we want to know how many reads fall into this region?
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools view sample_sorted.bam 20:20000-30000 | wc -l</li>
    </ul>
  </div>
  <div class='description'>
    Remember to not include the header in the above command, since the <span class="code-list code-span panel panel-danger">wc -l</span> command just counts lines. We want to count the number of alignments and not the total number of lines in the BAM file.
  </div>

  <!-- BAM flag -->
  <div class='section-subtitle'>The BAM flag</div>
  <div class='description'>
    The BAM flag (the second column) encodes lots of useful information, but isn't particularly readable by humans. We can use <i>samtools</i> to convert the bitwise flag into a list of characters. This is more readable than the simple number, but is still not great. A better way of viewing the meaning of the flag is <a href="https://broadinstitute.github.io/picard/explain-flags.html">here</a>, where you can type in the number and see a full explanation of the flag.
  </div>
  <div class='description'>
    What if we want to look at all the reads in the BAM file that are marked as being in a pair. We can use the following command to do this:
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools view -f 0x1 sample_sorted.bam | less -S</li>
    </ul>
  </div>
  <div class='description'>
    You will see that most of the reads have flags '99' or '163'. If we look at the flag explanations <a href="https://broadinstitute.github.io/picard/explain-flags.html">here</a>, we can see that both of these flags include the 1 (or 0x1) bit referring to the reads being paired.
  </div>

  <!-- samtools tview -->
  <div class='section-subtitle'>samtools tview</div>
  <div class='description'>
    <a href="https://www.broadinstitute.org/igv/"><i>IGV</i></a> is a very useful tool for looking at alignments together, but sometimes you just want a quick look at the data in a particular region. For example, we might want to see all the reads spanning a particular exon. Let's just quickly take a look at the reads that fall into the range 71,800 - 71,900 on chromosome 20.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools tview -p 20:71800-71900 sample_sorted.bam chr20_fragment.fa</li>
    </ul>
  </div>
  <div class='description'>
    We can clearly see an insertion of the sequence 'ACCAGT' in most of these reads. Lower case letters (or ',') represent alignment to the reverse strand, whereas uppercase letters (and '.') represent alignment to the forward strand. Press <span class="code-list code-span panel panel-danger">?</span> to view options for changing the view in <i>tview</i>, for example changing the colour scheme, or showing all bases, not just those that differ from the reference. Press <span class="code-list code-span panel panel-danger">q</span> to quit out of <i>tview</i>.
  </div>

  <!-- Clean up -->
  <div class='section-subtitle'>Clean up</div>
  <div class='description'>
    Having completed the tutorial, we can clean up all the files we created.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>cd ..</li>
      <li>rm -fr bam-demo</li>
    </ul>
  </div>
  <div class='description'>
    The flag <span class="code-list code-span panel panel-danger">-r</span> performs a recursive delete which is required for deleting directories.
  </div>
</div>
