---
layout: page
permalink: cshl_practical.html
---

<!-- -->
<div class='section'>
  <div class='section-title'>Advanced Sequencing Technologies and Applications course: Variant discovery practical</div>
  <div class='description'>
    This tutorial will starts with a set of FASTQ files and a reference genome FASTA file and walks through all necessary steps to generate alignments, variant calls and interrogate those calls.
  </div>

  <!-- Preparation -->
  <div class='section-subtitle'>Set up</div>
  <div class='description'>
    To begin, create a new directory call 'cshl-practical' and copy the necessary FASTQ files to this directory.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>cd workspace</li>
      <li>mkdir cshl-practical</li>
      <li>cd cshl-practical</li>
      <li>mkdir reference</li>
      <li>mkdir analysis</li>
      <li>REF="/home/ubuntu/workspace/data/fasta/GRCh37/Homo_sapiens.GRCh37.75.dna_sm.primary_assembly"</li>
      <li>DATA="/home/ubuntu/workspace/data/utah/"</li>
      <li>cd reference</li>
    </ul>
  </div>
  
  <!-- Explanation of some Unix stuff -->
  <div class='section-subtitle'>Some Unix stuff</div>
  <div class='description'>
    In case this is new to people, we are using a variable to save ourselves some time with writing out long file paths. Above, we defined the variable <span class="code-list code-span panel panel-danger">REF</span> with the path to the reference genome we will be using. In order to use this variable in a command, we need to prepend the variable with a dollar sign. Try it:
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>echo $REF</li>
    </ul>
  </div>
  <div class='description'>
    This command will evaluate the variable <span class="code-list code-span panel panel-danger">REF</span> and write its value to the screen.
  </div>
  
  <!-- Prepare the reference sequence -->
  <div class='section-subtitle'>Prepare the reference sequence</div>
  <div class='description'>
    We have the reference FASTA file, but aligners don't necessarily work with this file directly. We will be using <a href="https://github.com/lh3/bwa.git"><i>bwa</i></a> to perform the alignment, and this requires the reference to be in a particular format, accompanied by an array of index files. Bwa does have all the tools necessary to build these files itself, so first let's look at the options that bwa provides.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bwa</li>
    </ul>
  </div>
  <div class='description'>
    This lists all of the options available to bwa. You can see that the option <span class="code-list code-span panel panel-danger">index</span> handes the performs the indexing function on the FASTA file, so this is what we need to use. If we just type this command, we will see what additional information we need to provide.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bwa index</li>
    </ul>
  </div>
  <div class='description'>
    So we need to provide two things. First the prefix that will be used for all of the output files which we set as the same name as the FASTA files, and the FASTA file itself. So let's create the references indices.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bwa index -p chr20_fragment chr20_fragment.fa</li>
    </ul>
  </div>
  <div class='description'>
    Now take a look in this directory (<span class="code-list code-span panel panel-danger">ls</span>) and you will see a whole set of new files that will let us use <a href="https://github.com/lh3/bwa.git"><i>bwa</i></a> for aligning our FASTQ files.
  </div>

  <!-- Align the reads -->
  <div class='section-subtitle'>Align reads</div>
  <div class='description'>
    We will now go ahead and align the reads using <a href="https://github.com/lh3/bwa.git"><i>bwa</i></a>. We will be using the <i>mem</i> option within <a href="https://github.com/lh3/bwa.git"><i>bwa</i></a> to generate split read alignments. In reality, aligning reads is a multistep process, but we will build this up step by step. Some of these steps can be avoided by using workflows provided by different vendors, but for the purposes of learning the process, we will work through the whole process.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>cd ../analysis</li>
      <li>bwa mem</li>
    </ul>
  </div>
  <div class='description'>
    This shows us all of the command line arguments available to <a href="https://github.com/lh3/bwa.git"><i>bwa mem</i></a>. We won't go through all of the options, but will focus on some of the important ones. In particular, we need to specify the indexed reference genome and the <span class="code-list code-span panel panel-danger">fastq</span> files containing the reads to align.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bwa mem -t 8 -R "@RG\tID:NABOB\tPL:ILLUMINA\tSM:NA12878" ../reference/chr20_fragment $DATA/NA12878_1_sample.fastq $DATA/NA12878_2_sample.fastq</li>
    </ul>
  </div>
  <div class='description'>
    You probably noticed that a lot of stuff just got written to the screen. This isn't that helpful, since we need all of this to be output to a file. We could just redirect the output to a file:
  </div>
  <div class='description'>
    If we look at what we just created, we can see that the output files have long, cumbersome names. This is because <i>gkno</i> wasn't given any output names so built the names based on the input files and the tasks that were performed. It would be nicer to have more succinct names, so let's redo thie alignment (we can get away with doing this because these files are so small. Don't do this with real files!!!).
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bwa mem -t 8 -R "@RG\tID:NABOB\tPL:ILLUMINA\tSM:NA12878" ../reference/chr20_fragment $DATA/NA12878_1_sample.fastq $DATA/NA12878_2_sample.fastq > output.sam</li>
    </ul>
  </div>
  <div class='description'>
    Ok. Now we have a <span class="code-list code-span panel panel-danger">SAM</span> file, since it is not compressed into a binary form. We can convert this into a <span class="code-list code-span panel panel-danger">BAM</span> using samtools.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools view -b -S -o output.bam output.sam</li>
    </ul>
  </div>
  <div class='description'>
    Can you figure out what the command line options we just used are for?
    If we look at the <span class="code-list code-span panel panel-danger">SAM</span> file we just created, what do we notice?
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>less -S output.sam</li>
    </ul>
  </div>
  <div class='description'>
    All of the records are not in order. We see some reads that aren't mapped at all (remember we are using a reference genome that only represents a fraction of chromosome 20), then reads that map to chromosome 20, but in a random order. This is because the reads are output in the order they appeared in the <span class="code-list code-span panel panel-danger">FASTQ</span> file. So we need to sort this file.
  </div>
  <div class='description'>
    We could just go ahead and sort the <span class="code-list code-span panel panel-danger">BAM</span> file, but you'll notice that we are starting to build up a lot of files that we don't need. We will ultimately need to remove all of these files. So, what would be easier would be to build a command line that does all of these actions at once. Before we do that, let's remove the files we've already created:
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>rm -f output.*</li>
    </ul>
  </div>
  <div class='description'>
    So now let's run a command to generate what we need, and then walk through what it all means.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bwa mem -t 8 -R "@RG\tID:NABOB\tPL:ILLUMINA\tSM:NA12878" ../reference/chr20_fragment $DATA/NA12878_1_sample.fastq $DATA/NA12878_2_sample.fastq |</li>
      <li>samblaster |</li>
      <li>samtools view -b -S - |</li>
      <li>samtools sort -o output.bam -</li>
    </ul>
  </div>
  <div class='description'>
    There are a couple of useful things in the previous command line. The <span class="code-list code-span panel panel-danger">|</span> symbol (called a pipe), creates a 'stream'. This means that as data is generated by one tool, it passes this on to the next tool without making an intermediate file on the way. So the command aligns reads to the reference and passes these mapped reads on to the samblaster tool for marking duplicate reads. This, then passes the marked reads on to samtools to first convert the <span class="code-list code-span panel panel-danger">SAM</span> format into <span class="code-list code-span panel panel-danger">BAM</span> format before sorting the reads. At the end of all of this, we have a single file, <span class="code-list code-span panel panel-danger">output.bam</span>.
  </div>
  <div class='description'>
    One benefit to the compressed <span class="code-list code-span panel panel-danger">BAM</span> is that we can index it. This means that we can then ask for reads from a specific genomic location and we can easily jump to this location to extract the reads. We will use samtools again to index the <span class="code-list code-span panel panel-danger">BAM</span> file.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>samtools index output.bam -</li>
    </ul>
  </div>
  <div class='description'>
    This produced the index file <span class="code-list code-span panel panel-danger">output.bam.bai</span> that will allow us to jump around the corresponding <span class="code-list code-span panel panel-danger">BAM</span> file at will.
  </div>
  <div class='description' style="padding-top: 15px;">
    Now let's align the reads to the full reference, for three samples: NA12878, NA12891, and NA12892. In order to do these is parallel (e.g. at the same time), we can use 'screens'.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>rm -f *</li>
      <li>bwa mem -t 8 -R "@RG\tID:NA12878\tPL:ILLUMINA\tSM:NA12878" $REF $DATA/NA12878_1_sample.fastq $DATA/NA12878_2_sample.fastq |</li>
      <li>samblaster |</li>
      <li>samtools view -b -S - |</li>
      <li>samtools sort -o NA12878.bam -</li>
    </ul>
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bwa mem -t 8 -R "@RG\tID:NA12891\tPL:ILLUMINA\tSM:NA12891" $REF $DATA/NA12891_1_sample.fastq $DATA/NA12891_2_sample.fastq |</li>
      <li>samblaster |</li>
      <li>samtools view -b -S - |</li>
      <li>samtools sort -o NA12891.bam -</li>
    </ul>
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bwa mem -t 8 -R "@RG\tID:NA12892\tPL:ILLUMINA\tSM:NA12892" $REF $DATA/NA12892_1_sample.fastq $DATA/NA12892_2_sample.fastq |</li>
      <li>samblaster |</li>
      <li>samtools view -b -S - |</li>
      <li>samtools sort -o NA12892.bam -</li>
    </ul>
  </div>
  <div class='description'>
    Now we need to index the <span class="code-list code-span panel panel-danger">BAM</span> files. Let's use some more Unix magic to do this.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      for file in *.bam; do samtools index $file; done
    </ul>
  </div>
  <div class='description'>
    This is a standard programming 'loop'. The command <span class="code-list code-span panel panel-danger">for file in *.bam</span> sets up a loop over all files in the directory that end in <span class="code-list code-span panel panel-danger">.bam</span>. The asterisk is a wild card which can take any value. So the three <span class="code-list code-span panel panel-danger">BAM</span> files we just created match this pattern. Now, <span class="code-list code-span panel panel-danger">file</span> is a variable which takes the value of each file and applies the next commands to each file in turn. In this case, we index the bam file with samtools. We then include <span class="code-list code-span panel panel-danger">done</span> to terminate the loop. So, this command will loop over each <span class="code-list code-span panel panel-danger">BAM</span> file in turn and index it. If you now look at the contents of the directory, you will see the three <span class="code-list code-span panel panel-danger">BAM</span> files, each with a corresponding index (<span class="code-list code-span panel panel-danger">.bai</span>) file.
  </div><div class='description'>
    Let's generate statistics on all of the <span class="code-list code-span panel panel-danger">BAM</span> files in the same way. It is left to you to figure out the extra stuff in the following command!
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      for file in *.bam; do output=${file/bam/stats}; samtools flagstat $file > $output; done
    </ul>
  </div>

  <!-- Interrogate the BAM files -->
  <div class='section-subtitle'>Check the BAM files</div>
  <div class='description'>
    The first thing we can do is just look at the alignment statistics.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>less -S NA12878.stats</li>
    </ul>
  </div>
  <div class='description'>
    We can see that the alignments are good! Over 99% of the reads are mapped. We also have a very low number of duplicate reads, which is good for whole genome sequence data. But what is our average depth? We can get at this data using <a href="http://bam.iobio.io" target="_blank"><i>iobio</i></a>. If you are working on a local computer, use the 'choose bam file' button and navigate to the data. You will need to select both the BAM file and its index together. If working on the cloud, you can use the 'choose bam url' to specify the url of one of the BAM files we just created. 
  </div>
  <div class='description' style="padding-top: 15px;">
    Since the BAM files we created are very small and only have coverage in a couple of places in the genome, there aren't enough reads to generate statistics genome-wide. We can use the top 'Read Coverage' panel to select chromosome 20 and the zoom in on the area where alignments exist. We now generate statistics including the percentage of mapped reads etc in the region. We can now say that we have an average coverage of ~50X.
  </div>

  <!-- Optional bam.iobio screenshot -->
  <div class="panel-group" id="accordion" role="tablist" aria-multiselectable="true">
    <div class="panel panel-info">
      <div class="panel-heading" role="tab" id="panelHeading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#bamIobio" aria-expanded="false" aria-controls="bamIobio">
            bam.iobio screenshot
          </a>
        </h4>
      </div>
      <div id="bamIobio" class="panel-collapse collapse" role="tabpanel" aria-labelledby="panelHeading">
        <img src="/public/images/cshl15/bam.iobio.png" style="width: 1050px; padding-left: 10px; padding-right: 10px;">
      </div>
    </div>
  </div>

  <div class='description'>
    Just for comparison, let's look at a regular whole genome BAM file. Reload the <a href="http://bam.iobio.io" target="_blank"><i>bam.iobio</i></a> web page and click 'choose bam url' and click 'Go' on the default file. This is a 1000 Genomes BAM file for the NA12878 sample. <a href="http://bam.iobio.io"><i>bam.iobio</i></a> is sampling the BAM file in real time and generating the statistics based on this sampling. Instead of waiting for possibly hours to generate these statistics, we have them in seconds. We can get a good sense of important statistics like duplicate rates and quality score distributions immediately using this app.
  </div>
  <div class='description' style="padding-top: 15px;">
    We can also use <a href="http://bam.iobio.io" target="_blank"><i>bam.iobio</i></a> to look at exome BAM files. Reload the web page, click 'choose bam url' and add the url <span class="code-list code-span panel panel-danger">https://s3.amazonaws.com/iobio/1000gSV/NA19238.mapped.ILLUMINA.bwa.YRI.exome.20130415.bam</span>, then click 'Go'. We see the same statistics as before, but the 'Read Coverage Distribution' is not that useful. Since the majority of the genome has no coverage from exome sequencing, the distribution is heavily weighted towards very low read counts. To generate statistics only in relevant regions, you can upload a BED file, or just use the default BED supplied. Do this by clicking 'Default Bed' in the upper right corner of the top 'Read Coverage' panel. Now the 'Read Coverage Distribution' looks more like what we'd expect.
  </div>

  <!-- Variant calls -->
  <div class='section-subtitle'>Call variants</div>
  <div class='description'>
    Now let's use <i>freebayes</i> to call variants on our BAM files.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>freebayes -h</li>
    </ul>
  </div>
  <div class='description'>
    There are a huge number of commands available for <a href="https://github.com/ekg/freebayes.git"><i>freebayes</i></a>, but we can safely use the default values. Typically, we break calling up into regions to make this quicker. We will not dig too deeply into options here, so we will just call on a few chromosomes. To run things in parallel, we will use screens.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 17 -v chr17.vcf</li>
      <li>freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 20 -v chr20.vcf</li>
      <li>freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 21 -v chr21.vcf</li>
      <li>freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 22 -v chr22.vcf</li>
    </ul>
  </div>
  <div class='description'>
    Now we need to merge of all these files together to generate a single VCF file, compress and index. We will use bcftools and bgzip for these roles. The bcftools merge command requires the files to be compressed already, but we can use bcftools concat instead.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bcftools concat -o all.vcf chr17.vcf chr20.vcf chr21.vcf chr22.vcf</li>
      <li>bgzip all.vcf</li>
      <li>bcftools index all.vcf.gz</li>
    </ul>
  </div>
  <div class='description'>
    To review the format of VCF files, click the link below.
  </div>

  <!-- Optional vcf screenshot -->
  <div class="panel-group" id="accordion" role="tablist" aria-multiselectable="true">
    <div class="panel panel-info">
      <div class="panel-heading" role="tab" id="vcfHeading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#vcfFile" aria-expanded="false" aria-controls="vcfFile">
            vcf file review
          </a>
        </h4>
      </div>
      <div id="vcfFile" class="panel-collapse collapse" role="tabpanel" aria-labelledby="vcfHeader">
        <img src="/public/images/cshl15/vcf.png" style="width: 1080px; padding-left: 10px; padding-right: 10px;">
        <div class='section-subtitle'>Header</div>
        <div class='description'>
          The header describes many aspects of the contents of a VCF file. The INFO and FORMAT lines are particularly important as they describe how information is encoded in each VCF record for the variant in general (INFO) and the samples (FORMAT). For example, the allele frequency of each variant is described by the 'AF' attribute in each variant record.
        </div>
        <div class='section-subtitle'>Variant coordinate (CHROM and POS)</div>
        <div class='description'>
          For each variant record, the first two columns indicate the genomic coordinate of the variant.
        </div>
        <div class='section-subtitle'>Variant id (ID)</div>
        <div class='description'>
          The third column shows the rsid (reference SNP id). This is the generated by comparing the VCF with the dbSNP database, and is absent unless the VCF file has been specifically annotated with this information.
        </div>
        <div class='section-subtitle'>Alleles (REF and ALT)</div>
        <div class='description'>
          The REF (reference) allele is the allele as it appears in the reference genome which was used for variant calling. The length of the reference allele is dependent on the type of variant being described. A SNP will have a 1bp reference allele, whereas a reference relative deletion will have have a reference allele that encompasses all of the deleted bases in the mutation.
        </div>
        <div class='description' style="padding-top: 15px;">
         The ALT (alternate allele) is a comma separated list of all variants seen at this locus. It is possible that there are multiple alternate alleles; for example, consider microsatellites. We will discuss multi-allelic sites and complex alleles later in this tutorial.
        </div>
        <div class='section-subtitle'>Variant quality (QUAL)</div>
        <div class='description'>
          The PHRED score for the variant. Often, these scores are not well calibrated and do not have the probabilistic interpretation usually ascribed to PHRED values. But, higher quality variants will generally have larger quality scores, so these are still of value when trying to filter out low quality variants.
        </div>
        <div class='section-subtitle'>Filter (FILTER)</div>
        <div class='description'>
          Often, post-processing on VCF files is performed to filter out variants with specific properties. After such filtering, the FILTER field is populated with values such as 'PASS' to indicate passing variants, or, e.g. QUAL&lt;2, ro indicate variants that failed the specified filter.
        </div>
        <div class='section-subtitle'>Information on the variant (INFO)</div>
        <div class='description'>
          A host of key, value pairs in the form 'KEY=VALUE'. In a well-formed VCF (not all VCF files are!), each KEY will have an INFO line in the header explaining what it describes and the format of the value. There are a set of reserved KEYS (e.g. DP which describes the depth across all samples).
        </div>
        <div class='section-subtitle'>Sample information description (FORMAT)</div>
        <div class='description'>
          The FORMAT field describes the information that is stored for each sample in the final columns of the variant record, in the form A:B:C etc. Each key (A, B, C etc.) should have a FORMAT line in the header describing the information it encodes. 
        </div>
        <div class='section-subtitle'>Sample information</div>
        <div class='description'>
          Every sample in the VCF file is given a column containing the information described in the FORMAT line.
        </div>
      </div>
    </div>
  </div>

  <!-- Interrogate the VCF files -->
  <div class='section-subtitle'>Check the VCF files</div>
  <div class='description'>
    The quickest way to get a global view of our variant calls is to use <a href="http://vcf.iobio.io" target="_blank"><i>vcf.iobio</i></a>. In your web browser, go to the <a href="http://vcf.iobio.io" target="_blank"><i>vcf.iobio</i></a> website and select 'choose vcf file' (or 'choose vcf url' if your file is on the cloud). Select the compressed VCF file (this will have the extension <span class="code-list code-span panel panel-danger">vcf.gz</span>) and the index file (with extension <span class="code-list code-span panel panel-danger">vcf.gz.tbi</span>) together.
  </div>

  <!-- Optional vcf.iobio screenshot -->
  <div class="panel-group" id="accordion" role="tablist" aria-multiselectable="true">
    <div class="panel panel-info">
      <div class="panel-heading" role="tab" id="vcfIobioHeading">
        <h4 class="panel-title">
          <a role="button" data-toggle="collapse" data-parent="#accordion" href="#vcfIobio" aria-expanded="false" aria-controls="vcfIobio">
            vcf.iobio screenshot
          </a>
        </h4>
      </div>
      <div id="vcfIobio" class="panel-collapse collapse" role="tabpanel" aria-labelledby="vcfIobioHeading">
        <img src="/public/images/cshl15/vcf.iobio.png" style="width: 1050px; padding-left: 10px; padding-right: 10px;">
      </div>
    </div>
  </div>
  <div class='description'>
    The application shows information only in chromosomes 17, 20, 21, and 22, and also shows that the density of variants is restricted to very small regions. If we knew nothing about this VCF file until now, we get a great deal of information from this simple view of the data. This is a great way to see if the data is what you expected it to be. Is the sample supposed to be female, but has lots of calls on the Y chromosome? Is this supposed to be whole genome data, but large regions of the genome have no variants? The information supplied here gives a quick and easy way to ask these questions.
  </div>
  <div class='description' style="padding-top: 15px;">
    To see <a href="http://vcf.iobio.io"><i>vcf.iobio</i></a> showing information on the entire <a href="http://exac.broadinstitute.org/"><i>ExAC</i></a> database, click <a href="http://vcf.iobio.io/?vcf=http://s3.amazonaws.com/vcf.files/ExAC.r0.2.sites.vep.vcf.gz" target="_blank">here</a>. It is clear to see features like the centromere on chromosome 1 where no calls have been made. Also, metrics like allele frequency and the distribution of base changes are available.
  </div>

  <!-- VCF manipulation -->
  <div class='section-subtitle'>Manipulate VCF files</div>
  <div class='description'>
    We will now manipulate the VCF files using bcftools. This will perform the majority of the functions we need, although other tools exist, including <a href="http://genome.sph.umich.edu/wiki/Vt"><i>vt</i></a> and <a href="https://github.com/ekg/vcflib"><i>vcflib</i></a>.
  </div>
  <div class='description'>
    First, let's view a specific portion of our VCF file, e.g. chromosome 17 only.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bcftools view all.vcf.gz 17 | less -S</li>
    </ul>
  </div>
  <div class='description'>
    Use the space bar and <span class="code-list code-span panel panel-danger">u</span> to navigate up and down through the VCF file by pages, or the up and down cursor keys to move up and down line by ine. The left and right cursor keys will let you move left and right through the whole VCF records. Let's look at a couple of variants that may need to be modified depending on your analysis goals. First, we can pull out all 'snp' alleles from the VCF file..
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bcftools view -i 'TYPE="snp"' all.vcf.gz | less -S</li>
    </ul>
  </div>
  <div class='description'>
    Let's look at one more record.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bcftools view all.vcf.gz 22:39636944 | less -S<li>
    </ul>
  </div>
  <div class='description'>
    By scrolling to the right, we can see that the three samples have genotypes '1/2', '1/2' and '1/3'. So all of our samples are heterozygous and none of them have the reference allele in any chromosome. '1' refers to the first alternate allele, '2' the second alternate etc. We can break these multi-allelic sites into multiple lines - one for each allele.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bcftools view all.vcf.gz 22:39636944 | </li>
      <li>bcftools norm -m - </li>
      <li>| less -S</li>
    </ul>
  </div>
  <div class='description'>
    After we ran this command, we see that the single multi-allelic variant was broken into three seperate lines, all at the same genomic coordinate. You may notice that the presentation of the alleles is unusual. The first variant is a deletion, but the second variant is SNP, but the base that is mutated is the last base in a five base sequence. If we perform a comparison between different VCF files, representing a SNP in this way will almost certainly mean that even if the two files being compared share the SNP, the comparison would fail. It is extremely important that when comparing VCF files, we have a common, unambiguous representation of the alleles. We can <i>normalize</i> the variants to achieve this. In this case, that just means including the reference genome in the command line, so that these new variants can be interpreted.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>bcftools view all.vcf.gz 22:39636944 | </li>
      <li>bcftools norm -f $REF.fa -m - </li>
      <li>| less -S</li>
    </ul>
  </div>
  <div class='description'>
    Now we have a simple deletion and two normal looking SNPs! Normalizing variants is an important step that should not be forgotten!
  </div>
    
  <!-- Subsetting on samples -->
  <div class='section-subtitle'>Subsetting on samples</div>
  <div class='description'>
    If we perform joint calling with a large population of samples, we will often have a VCF file that contains hundreds or thousands of samples. But what if we're only interested in one or two of these samples?
  </div>
  <div class='description' style="padding-top: 15px;">
    There are two useful tools that we might want to use here, depending on our ultimate goal. The first tool, vcflib, will just keep the samples that we want, but not change the number of variant records in the file.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>$VCFLIB/vcfkeepsamples all.vcf.gz NA12878 > NA12878.vcf</li>
      <li>gkno compress-vcf -i NA12878.vcf</li>
    </ul>
  </div>    
  <div class='description'>
    We have generated the new VCF file, NA12878.vcf.gz, which is exactly the same as all.vcf.gz, except that there is only one sample column: that for the sample NA12878. We could have listed as many samples as we liked in the command line, and each of those would have been kept.
  </div>
  <div class='description' style="padding-top: 15px;">
    What is the problem with this? The main problem here, is that we often want a VCF file that contains information about the requested samples, but only keeps variants that are polymorphic in those samples. For example, let's take a look at this new file in a bit more detail.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>less NA12878.vcf.gz | grep "0/0" | less -S</li>
    </ul>
  </div> 
  <div class='description'>
    Every one of these lines is a variant that the sample NA12878 does not have - the genotype is 0/0, or homozygous reference. So what if we want to only keep the sample information for NA12878, but additionally, remove all homozygous reference variants? 
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>echo "NA12878" > samples.txt</li>
      <li>vt subset -s samples.txt all.vcf.gz -o NA12878_only.vcf.gz</li>
      <li>less NA12878.vcf.gz | grep -v "^#" | wc -l</li>
      <li>less NA12878_only.vcf.gz | grep -v "^#" | wc -l</li>
    </ul>
  </div> 
  <div class='description'>
    With those last bash commands, we can see that the NA12878_only.vcf.gz file has 119 less variants than NA12878.vcf.gz. Let's just check that that makes sense.
  </div>
  <div class="panel panel-danger">
    <ul class="code-list">
      <li>less NA12878.vcf.gz | grep "0/0" | wc -l</li>
    </ul>
  </div>
  <div class='description'>
    Exactly 119 variants kept in NA12878.vcf.gz were homozygous reference and were removed by using the vt subset command. Again, we can subset on multiple samples by adding more sample IDs to the samples.txt file.
  </div>
</div>
