---
layout: post
title: ACMG 56 genes
image: /public/images/blog/acmg56/main.png
subtitle: Analyzing genes identified by the ACMG
tags:
  - Alistair Ward
  - gene.iobio
---
In March 2013, an <a href='https://www.acmg.net/' target='_blank'>The American College of Medical Genetics and Genomics (ACMG)</a> working group released a list of 57 genes (later revised to 56), that harbour mutations that when identified, could lead to severe outcomes, but were medically actionable. These genes are implicated in a number of disorders, including inherited cancers, inherited cardiac diseases, connective tissue diseases affecting the cardiovascular system, familial hypercholesterolemia, and malignant hyperthermia susceptibility. The ACMG also generated a list of <a href='http://www.ncbi.nlm.nih.gov/pubmed/23788249' target='_blank'>recommendations</a> that state that known pathogenic (or in some cases expected pathogenic) variants discovered in any of these genes should be reported regardless of an individuals age. These recommendations referred to what are now known as secondary findings in an individual undergoing exome or genome sequencing.

It was felt that ethical concerns around testing minors for secondary findings was compensated by the potential benefit to the future health of the individual and their family. As such, the working group did not support an option to allow an opt-out of receiving this information, however, in <a href='http://www.nature.com/gim/journal/v17/n1/full/gim2014151a.html' target='_blank'>March 2014</a>, the recommendations were relaxed to allow an opt-out. It was then recommended that individuals undergoing exome or genome sequencing should undergo counseling before and after testing. Annual updates to the ACMG policy on secondary findings is expected as sequencing becomes more ubiquitous and our ability to interpret findings improves.

The importance of these genes is clear, and it is likely that when analyzing an individual or a family in <a href='http://gene.iobio.io' target='_blank'>gene.iobio</a>, looking at variants in them will be desirable. To make this an easy task for the user, we have made the ACMG56 genes a default list that can be easily selected for analysis. If we open up <a href='http://gene.iobio.io/?rel0=proband&rel1=mother&rel2=father&name0=NA19240&vcf0=https://s3.amazonaws.com/iobio/gene/wes_1000g/exome-trio.vcf.gz&bam0=https://s3.amazonaws.com/iobio/gene/wes_1000g/NA19240.bam&sample0=NA19240&name1=NA19238&vcf1=https://s3.amazonaws.com/iobio/gene/wes_1000g/exome-trio.vcf.gz&bam1=https://s3.amazonaws.com/iobio/gene/wes_1000g/NA19238.bam&sample1=NA19238&name2=NA19239&vcf2=https://s3.amazonaws.com/iobio/gene/wes_1000g/exome-trio.vcf.gz&bam2=https://s3.amazonaws.com/iobio/gene/wes_1000g/NA19239.bam&sample2=NA19239&gene=RAI1&genes=RAI1,AIRE,PDGFB,MYLK2,PDHA1' target='_blank'>gene.iobio</a> with the modified 1000 Genomes exome trio dataset loaded, click on \'Genes\' at the left of the screen, we can then select \'Import ACMG56 Genes\'.

<img src="/public/images/blog/acmg56/select.png" style="width:55%; margin:20px 250px;">

Once selected, the ACMG56 genes replace all other genes that were being analyzed and appear as gene badges (as discussed in the <a href='http://127.0.0.1:4001/2016/03/28/multi-gene/' target='_blank'>previous post</a>). If we then select \'Analyze all\', each of the genes is analyzed in turn. By clicking on the gene badge, we can quickly jump to any of the analyzed genes and assess the contained variants, while processing continues through the list. If we perform the analysis on the ACMG56 genes for this modified exome trio dataset, we see the following:

<img src="/public/images/blog/acmg56/genes.png" style="width:100%; margin:20px 0px;">

With almost no effort on our part, we have scanned for secondary findings in this individual. The results suggest that we should take a closer look at BRCA2, APC, MYBPC3, TMEM43, SCN5A, PCSK9 and CACNA1S. If we take a look in the APC gene, for example, we can see that the child has a heterozygous missense variant inherited from her mother that is listed as of uncertain significance in ClinVar, but is predicted to be possibly_damaging by PolyPhen. The variant has very low allele frequency in the 1000 Genomes data (0.02%) and is not seen in ExAC.

<img src="/public/images/blog/acmg56/APC.png" style="width:70%; margin:20px 100px;">

Being able to quickly analyze the ACMG56 genes gives a quick and easy way to analyze the proband in a set of genes known to be implicated in a number of diseases and have potentially actionable outcomes. This is an extremely powerful feature which we really hope the community finds useful.
